module-system/module-implementations/sov-ibc-transfer/src/call.rs

```diff
@@ -0,0 +1,131 @@
+use ibc::core::ExecutionContext;
+use ibc::Signer;
+use sov_state::WorkingSet;
+
+use crate::context::EscrowExtraData;
+use crate::Transfer;
+
+#[derive(borsh::BorshDeserialize, borsh::BorshSerialize, Debug, PartialEq)]
+pub struct SDKTokenTransfer<C: sov_modules_api::Context> {
+    /// the port on which the packet will be sent
+    pub port_id_on_a: PortId,
+    /// the channel by which the packet will be sent
+    pub chan_id_on_a: ChannelId,
+    /// Timeout height relative to the current block height.
+    /// The timeout is disabled when set to None.
+    pub timeout_height_on_b: TimeoutHeight,
+    /// Timeout timestamp relative to the current block timestamp.
+    /// The timeout is disabled when set to 0.
+    pub timeout_timestamp_on_b: Timestamp,
+    /// The address of the token to be sent
+    pub token_address: C::Address,
+    /// The amount of tokens sent
+    pub amount: sov_bank::Amount,
+    /// The address of the token sender
+    pub sender: Signer,
+    /// The address of the token receiver on the counterparty chain
+    pub receiver: Signer,
+    /// Additional note associated with the message
+    pub memo: Memo,
+}
+
+impl<C> Transfer<C>
+where
+    C: sov_modules_api::Context,
+{
+    pub fn transfer(
+        &self,
+        sdk_token_transfer: SDKTokenTransfer<C>,
+        execution_context: &mut impl ExecutionContext,
+        token_ctx: &mut impl TokenTransferExecutionContext<EscrowExtraData<C>>,
+        working_set: Rc<RefCell<&mut WorkingSet<C::Storage>>>,
+    ) -> Result<sov_modules_api::CallResponse> {
+        let msg_transfer: MsgTransfer = {
+            let denom = {
+                let token_name = self
+                    .bank
+                    .get_token_name(
+                        &sdk_token_transfer.token_address,
+                        &mut working_set.borrow_mut(),
+                    )
+                    .ok_or(anyhow::anyhow!(
+                        "Token with address {} doesn't exist",
+                        sdk_token_transfer.token_address
+                    ))?;
+
+                if self.token_was_created_by_ibc(
+                    &token_name,
+                    &sdk_token_transfer.token_address,
+                    &mut working_set.borrow_mut(),
+                ) {
+                    // The token was created by the IBC module, and the ICS-20
+                    // denom was stored in the token name. Hence, we need to use
+                    // the token name as denom.
+                    token_name
+                } else {
+                    // This applies to all other tokens created on this
+                    // sovereign SDK chain. The token name is not guaranteed to
+                    // be unique, and hence we must use the token address (which
+                    // is guaranteed to be unique) as the ICS-20 denom to ensure
+                    // uniqueness.
+                    sdk_token_transfer.token_address.to_string()
+                }
```

Comment:
Thanks, this fixes the problem we talked about! 🙌 

I've got one more thing on my mind:
With this condition, wouldn't we end up with a mix of minted tokens, some coming from their names, and others from their addresses? Aside from that, this is not much user-friendly, It kind of goes against the whole idea of why the `SDKTokenTransfer` has been abstracted, letting us use token names as the denom.
People can create bunch of phony tokens with names similar to legit ones, or mess with the availability of the good names. So, we might find ourselves stuck with token names in the transfer module that are mostly based on addresses.